src/katgpucbf/xbgpu/kernels/tensor_core_correlation_kernel.cu

```diff
@@ -246,12 +247,14 @@ template <typename T> struct FetchData
   {
   }
 
-  __device__ void load(const Samples samples, unsigned channel, unsigned time, unsigned firstReceiver, bool skipLoadCheck = NR_RECEIVERS % NR_RECEIVERS_PER_BLOCK == 0)
+  __device__ void load(const Samples *samples, unsigned channel, unsigned time, unsigned firstReceiver, bool skipLoadCheck = NR_RECEIVERS % NR_RECEIVERS_PER_BLOCK == 0)
   {
     if (skipLoadCheck || firstReceiver + loadRecv < NR_RECEIVERS)
     {
-      data = * (T *) &samples[firstReceiver + loadRecv][channel][time][loadTime][0];
-      //memcpy(&data, &samples[firstReceiver + loadRecv][channel][time][loadTime][0], sizeof(T));
+      unsigned outerTime = time / NR_BLOCKS_PER_BATCH;
+      unsigned innerTime = time % NR_BLOCKS_PER_BATCH;
+      data = * (T *) &samples[outerTime][firstReceiver + loadRecv][channel][innerTime][loadTime][0];
```

Comment:
Once more, I know I could quickly hack away and/or look this up for my own clarity, but this review has been quite a journey already (and a good one) so I'm just going to ask here:
* Re: The update to require `*samples` instead of `samples` (in the arglist), what is the reasoning there?
  * I gather it's a practice in C++ that is ~better?
* Again, showing my naivety on C++, the `Samples` type doesn't seem to have explicitly _gained_ an extra dimension. Your documentation update pretty neatly says 
  * "The `FetchData` class splits the time index into a batch index and an offset within the batch [`outerTime` and `innerTime`], so that the rest of the code can be oblivious to batches, and just work in time "blocks" (`NR_TIMES_PER_BLOCK` spectra).
  * Obviously, brilliant. But how does the indexing "actually work" now? I don't expect anything like a Python IndexError; it's the addition of a new indexing dimension (`[outerTime]`) that's throwing me a bit.
  * You're still combing across the same amount of data, just in a different way.

Reply:
C is rather weird about arrays and particularly about passing arrays to functions. In various contexts (including parameter passing), an array "decays" to a pointer to the first element of the array. So now instead of passing a single instance of SampleData, I'm passing an array of SampleData by passing a pointer to the first element of the array.

Reply:
Noted and appreciated.

Still, a bit of a question mark around the updated indexing of `samples`.
* Are my points above valid?
* Slash, have I understood it mostly/~correctly around "the number of axes over which you index doesn't matter as long as you're within the bounds of the desired data"?